pkg/apply/apply.go

```diff
@@ -205,6 +209,33 @@ func NewTopicApplier(
 	}, nil
 }
 
+// helper function to check if there are any actual updates on the topic 
+// when returning applyExistingTopic
+func ensureChangesOccurred(updateChanges *UpdateChangesTracker) (*UpdateChangesTracker) {
+	replicaAssignmentsChanged := false
+	if updateChanges.ReplicaAssignments != nil {
+		for _, partition := range *updateChanges.ReplicaAssignments {
+			if partition.UpdatedReplicas != nil {
+				replicaAssignmentsChanged = true
+			}
+		}
+	}
+	if !replicaAssignmentsChanged {
+		updateChanges.ReplicaAssignments = nil
+	}
+
+	// if nothing actually changed, report changes as nil
+	if (updateChanges.NumPartitions == nil &&
+		updateChanges.NewConfigEntries == nil &&
+		updateChanges.UpdatedConfigEntries == nil &&
+		len(updateChanges.MissingKeys) == 0 &&
+		updateChanges.ReplicaAssignments == nil &&
+		!updateChanges.Error) {
+		return nil
+	}
+	return updateChanges
+}
```

Comment:
This logic was moved from the deleted `checkForUpdates()` below to here. Instead of checking if updates happened after every step of `applyExistingTopic()`, we just check in a single place in the function one level above `applyExistingTopic()`.